pkg/disk/filesystem.go

```diff
@@ -1,12 +1,57 @@
 package disk
 
 import (
+	"encoding/json"
+	"fmt"
 	"math/rand"
 	"reflect"
 
 	"github.com/google/uuid"
 )
 
+type MkfsOption int
+
+const ( // MkfsOption type enum
+	MkfsVerity MkfsOption = iota // Enable fs-verity option if needed (typically for EXT4)
+)
+
+func getMkfsOptionMapping() []string {
+	return []string{"verity"}
+}
+
+// ToString converts MkfsOption into a human readable string
+func (option MkfsOption) ToString() string {
+	return getMkfsOptionMapping()[int(option)]
+}
+
+func unmarshalHelper(data []byte, mapping []string) (int, error) {
+	var stringInput string
+	err := json.Unmarshal(data, &stringInput)
+	if err != nil {
+		return 0, err
+	}
+	for n, str := range mapping {
+		if str == stringInput {
+			return n, nil
+		}
+	}
+	return 0, fmt.Errorf("invalid mkfsoption: %s", stringInput)
+}
+
+// UnmarshalJSON converts a JSON string into an MkfsOption
+func (option *MkfsOption) UnmarshalJSON(data []byte) error {
```

Comment:
I know its a bit silly for trivial things like this but I do feel better having tests for these helpers, I did this boring work in https://github.com/osbuild/images/pull/1545